src/Slackbot.Net.Extensions.FplBot/Helpers/EntryForGameweek.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Fpl.Client.Abstractions;
+using Fpl.Client.Models;
+using Slackbot.Net.Extensions.FplBot.Abstractions;
+
+namespace Slackbot.Net.Extensions.FplBot.Helpers
+{
+    internal class EntryForGameweek: IEntryForGameweek
+    {
+
+        private readonly IEntryClient _entryClient;
+
+        public EntryForGameweek(IEntryClient entryClient)
+        {
+            _entryClient = entryClient;
+        }
+
+        public async Task<GameweekEntry> GetEntryForGameweek(ClassicLeagueEntry entry, int gameweek)
+        {
+            try
+            {
+                var entryPicksTask = _entryClient.GetPicks(entry.Entry, gameweek);
+                var entryPicks = await entryPicksTask;
```

Comment:
Can simply be `var entryPicks = await _entryClient.GetPicks(entry.Entry, gameweek);` when there's only one async call, and not several that needs to be fired away in parallel.